### ISSUE TITLE
NativeScript debugger was not able to get inspector socket port on device

### ISSUE THREAD

**Issue by marblsy** [NativeScript/NativeScript#9233]
**Environment**
Provide version numbers for the following components (information can be retrieved by running `tns info` in your project folder or by inspecting the `package.json` of the project):
 - CLI: 7.2.0
 - Cross-platform modules: 
 - Android Runtime: 7.0.1
 - iOS Runtime: 7.2.0
 - XCode Version: 12.4
 - Plugin(s): RadCalendar (7.2.0), RadListView (9.1.0)

**Describe the bug**
NativeScript debugger was not able to get inspector socket port on device F2028502-6846-4F76-8E81-90661EAE9440 for application org.nativescript.application.

On work we are using MacOS Catalina and my project works without any problems .
However at home I am using BigSur (M1 Chip) I cant get the inspector socket to work.

I cant remember exactly but I think it worked when I was using xCode 12.2, but after updating to 12.4 I get the above error. Using xCode 12.2 again now the inspector doesnt work as well. So I am stuck with a not working inspector socket and I am running out of ideas. I even created a completely fresh project, sadly same effect.

**Comment by marblsy** [comment]
Completely removed nativescript as well as xcode and reinstalled. No improvement

**Comment by marblsy** [comment]
This is down to the absolutlely bogged installation process of pip install six...
No matter what I set my pip path to, the ns doctor command and "configure for local builds" always reverts back to the 2.7 python which isnt even supported by six anymore

**Comment by marblsy** [comment]
Having completely wiped I ran into the problem once again which is getting increasingly frustrating.
pip install six isnt working on my mac anymore due to giving me a syntax error.
python -m pip install six however works as it is selecting the CORRECT and SUPPORTED version of pip to install six.
Sadly this doesnt fix the missing log. ns doctor returns no issues but this is just beyond satisfying and should be fixed asap.

**Comment by rigor789** [comment]
@marblsy if you are referring to the setup script - you can open a PR to change the command here: https://github.com/NativeScript/setup-scripts/blob/a93ac8ab8121f9657c36dae697e3d9f454a13016/native-script.rb#L222

We'll soon have an m1 mac to test on - but sadly it's impossible to fix issues related to m1 without having one. In these cases, we can only rely on the community (including yourself) to debug issues like these, and submit a fix.

**Comment by marblsy** [comment]
Understandable.
However I installed six now outside of the setup script and it gets recognized by ns doctor. I am still not able to get the inspector socket to work sadly

**Comment by marblsy** [comment]
I have now installed my mac completely new and followed the setup process again: Now I get the warning that my cocoapods are not configured properly despite the setup throws no errors :D I am geniunly out of ideas on how to get this fixed ever again because that is sth that worked with problems before

**Comment by vallemar** [comment]
I have a 2019 mac with an i7 and I am in the same situation.

**Comment by shirakaba** [comment]
> For the love of god if anyone ever tries to install nativescript: set your terminal to run in rosetta mode

@marblsy It may not be of any use at this point as it seems you're fully set up, but I took notes of [the setup steps I went through myself](https://gist.github.com/shirakaba/87aabee6e2ec70e298e3d7c42f701675). Yep, Rosetta mode is still the only way forward for now.

> NativeScript debugger was not able to get inspector socket port on device F2028502-6846-4F76-8E81-90661EAE9440 for application org.nativescript.application.

Yep, I experienced this on my (borrowed) M1 Mac Mini. You may also notice that all of your `console.log()` calls fail to produce any output in the terminal at all. I think these problems are related – to my understanding, the NativeScript CLI follows logs from Xcode and/or simulator to identify the port number, but I think the way it does that may need to change for M1. It may be as simple as updating a regex for reading logs from the NativeScript app's process.

I also found that basic things like `setState()` weren't working in my React NativeScript app. So I think this issue may be just the tip of the iceberg for NativeScript on Apple Silicon.

> I have a 2019 mac with an i7 and I am in the same situation.

@marblsy I think your issue is clearly due to developing on M1. I'll change the issue title to reflect that. @vallemar if you're experiencing the same problems, could you please file a new issue with details? I'd also recommend simply restarting your computer and seeing whether it self-resolves (issues involving ports can easily be like that).

**Comment by marblsy** [comment]
@shirakaba yeah, console.log isn't doing anything as well sadly. Regarding your mention of your setup could you give an update on the pip easy_install six as that isn't supported anymore and breaks when you follow the setup (due to using python 2.7 which isn't supported by six anymore)?

**Comment by shirakaba** [comment]
> could you give an update on the pip easy_install six as that isn't supported anymore and breaks when you follow the setup (due to using python 2.7 which isn't supported by six anymore)?

@marblsy I'm not familiar with that; the steps I wrote worked for me about a month ago, so this is a very new development. Could you perhaps explicitly specify the version of `six` that you want to install, to be the last one one supported by python 2.7? Other than that, not sure what to suggest (apart from doing those steps while the terminal is not in Rosetta mode, to match what I think I did).

**Comment by rigor789** [comment]
Perhaps it's better to not use the setup-script on m1 yet - and just install the requirements manually (that's what I usually do anyways, less "magical").

**Comment by mBaldasty** [comment]
@shirakaba @rigor789 (this is my other account): If you just use the setup process you cant alter which six version will be installed. It will pull the latest by default while the setup itself references pyhton 2.7. This combination results in a syntax error which is due to the six not supporting python 2.7 and lower.

Due to this I went to install all manually which (requires python -m pip install six) works without any problems. And said installed six version is recognized by the ns doctor setup as well. 

This was merely meant just for new guys who want to try nativescript out and rely on the setup, running into this problem and are faced with the first problem before even getting started

**Comment by mBaldasty** [comment]
Regarding the changed location of the logs: If anyone could give me the current location on the intel Macs and the related script that pulls them I could look into it the following days to hopefully provide something useful

**Comment by shirakaba** [comment]
> Regarding the changed location of the logs: If anyone could give me the current location on the intel Macs and the related script that pulls them I could look into it the following days to hopefully provide something useful.

@mBaldasty The relevant bits of code to parsing the logs, I believe, are these:

https://github.com/NativeScript/nativescript-cli/blob/25410b74a411bfa04d0aa968d47e2bd5a3c2cd1a/lib/services/ios-debugger-port-service.ts#L60-L73

https://github.com/NativeScript/nativescript-cli/blob/25410b74a411bfa04d0aa968d47e2bd5a3c2cd1a/lib/services/log-parser-service.ts#L8

https://github.com/NativeScript/nativescript-cli/blob/25410b74a411bfa04d0aa968d47e2bd5a3c2cd1a/lib/common/constants.ts#L64-L66

**Comment by nikoTM** [comment]
I do not think this is Apple Silicon exclusive, I am having the same problems on a regular mac, I logged it here https://github.com/NativeScript/nativescript-cli/issues/5437 It seems like it's related to XCode 12.4 or BigSur...I have wasted 2 days on trying to get the debugger to work on a blank project, still no solution for this. First it was saying "NativeScript debugger was not able to get inspector socket port on device", then after some reinstalls/updates/upgrades it started connecting again, but only shows Console/Sources, now I am back at not being able to connect (it crashes the app upon connection without errors), or when it does connect it will only sources ...I have no idea how else to approach this and it's a really annoying issue since it hard blocks iOS development.

**Comment by marblsy** [comment]
I wasn't able to run tests yet but that would go along with the feeling I had. I am sure it worked with Xcode 12.2. I then upgraded to 12.4 and it broke the debugging. Downgrading again sadly didn't resolve it

**Comment by marblsy** [comment]
Small update: When using a real device connected via usb the logging works without problems
